website_adv_image_optimization/controllers/main.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2018 Onestein
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import io
+import base64
+import json
+
+from odoo import http, tools
+from odoo.http import request
+from PIL import Image
+
+
+class MainController(http.Controller):
+    @http.route('/website_adv_image_optimization/attachment/optimize',
+                type='http', auth='user', methods=['POST'])
+    def optimize(self, attachment_id, quality=80,
+                 width=False, height=False, **kwargs):
+        attachment_obj = request.env['ir.attachment']
+        attachment = attachment_obj.search([('id', '=', attachment_id)])
+
+        image = Image.open(io.BytesIO(base64.b64decode(attachment.datas)))
```

Comment:
can't you use some tools from here https://github.com/odoo/odoo/blob/11.0/odoo/tools/image.py ?

Reply:
Not really, because those do 1 operation at the time it will be less efficient.

Reply:
ok, then 3 recommendations: 
1. add a comment stating it (for me, for your future self and the ones that will ask the same question ;))
2. move resizing to a separated method 
3. close IO objects after getting the value